python/risk_meters/historical_vuln_count.py

```diff
@@ -0,0 +1,187 @@
+    get_report_url = f"{base_url}asset_groups/{id}/report_query/{report}?start_date={start_date}"
+
+    # Invoke an Asset Group Report API.
+    response = requests.get( get_report_url, headers=headers)
+    if response.status_code != 200:
+        print(f"Asset Group Report API Error: {response.status_code} with {get_report_url}")
+        sys.exit(1)
+
+    resp_json = response.json()
+    return resp_json
+
+# Processes vulnerability count report and accepted risk report.
+def process_reports(base_url, headers, id, start_date, one_shot):
+    vuln_count_report = "historical_open_vulnerability_count_by_risk_level"
+    vuln_count_resp = get_historical_report(base_url, headers, id, vuln_count_report, start_date)
+
+    accepted_risk_report = "risk_accepted_over_time"
+    accepted_risk_resp = get_historical_report(base_url, headers, id, accepted_risk_report, start_date)
+
+    # Verify asset group IDs
+    if vuln_count_resp['id'] != accepted_risk_resp['id']: 
+        print(f"Reports don't have matching IDs: {vuln_count_report['id']}, {accepted_risk_resp['id']}")
+        sys.exit(1)
+    
+    if vuln_count_resp['id'] != id:
+        print(f"Reports don't matching requested ID: {vuln_count_report['id']}, {id}")
+        sys.exit(1)
+
+    # Get to data as a dictionary of dates with each element containing low, medium, and high.
+    vuln_count_data = vuln_count_resp['historical_vulnerability_count_by_risk']
+    risk_accepted_data = accepted_risk_resp['risk_accepted_over_time']
+
+    # Initialize output table.
+    vuln_count_tbl = PrettyTable()
+    vuln_count_tbl.field_names = ["Date", "High", "RA High", "Medium", "RA Medium", "Low", "RA Low"]
+
+    # Process both vulnerability count and risk accepted vulnerability count data.
+    for (vc_date, ra_date) in zip(vuln_count_data, risk_accepted_data):
+        if vc_date != ra_date:
+            print(f"Dates don't match: {vc_date} : {ra_date}")
+            sys.exit(1)
+
+        # Add today or the first of the month to the table.
+        if one_shot or vc_date[-2:] == "01":
+            vc_high = int(vuln_count_data[vc_date]['high'])
+            vc_medium = int(vuln_count_data[vc_date]['medium'])
+            vc_low = int(vuln_count_data[vc_date]['low'])
+            ar_high = int(risk_accepted_data[ra_date]['high'])
+            ar_medium = int(risk_accepted_data[ra_date]['medium'])
+            ar_low = int(risk_accepted_data[ra_date]['low'])
+   
+            vuln_count_tbl.add_row([vc_date, vc_high, ar_high, vc_medium, ar_medium, vc_low, ar_low])
+        
+    print(vuln_count_tbl)
+    print("")
+
+def print_help(program):
+    print(f"Prints out the historical risk meter vulnerability counts for a year.")
+    print(f"{program} <risk meter name> [today]")
+    print(f"   <risk meter name> - name of the risk meter")
+    print(f"   [today] - if present will only print the vulnerability counts for today")
+    print("")
```

Comment:
❤️ 

Do you want to mention [ risk_meter_id ] [ risk_meter_score ] [ risk_accepted_score ] as additional optional params?

Reply:
No, I did not.  Since it is only for the subprocess option.